maintenance/measure_tile_performance/measure_tile_performance.py

```diff
@@ -7,97 +7,135 @@
-    exit()
+    slug_input = args.slugInput
+    times = args.times
 
-# looks for environment variables for authentication
-sdk = looker_sdk.init40()
+    if args.sleepTimer:
+        sleep_timer = args.sleepTimer
+    else:
+        sleep_timer = 300
 
-query_list = []
-header = ['SlugId', 'Timestamp', 'RunTime']
+    if times > 100 or times <= 0:
+        print("Runs per query(arg2) must be between 1 and 100")
+        exit()
 
-from datetime import datetime
-# datetime object containing current date and time
-now = datetime.now()
- 
-# dd/mm/YY H:M:S
-filename = slug_input + now.strftime("_%Y%m%dT%H%M%S") + '.csv'
+    # looks for environment variables for authentication
+    sdk = looker_sdk.init40()
 
-# get query_id from slug
-slug_response = sdk.query_for_slug(slug=slug_input)
+    query_list = []
+    header = ['SlugId', 'Timestamp', 'RunTime']
 
-query = slug_response.id
+    from datetime import datetime
+    # datetime object containing current date and time
+    now = datetime.now()
+    
+    # dd/mm/YY H:M:S
+    filename = slug_input + now.strftime("_%Y%m%dT%H%M%S") + '.csv'
 
-if args.file:
-    with open(filename, 'w', encoding='UTF8') as f:
-        writer = csv.writer(f)
-        # write the header
-        writer.writerow(header)
+    # get query_id from slug
+    slug_response = sdk.query_for_slug(slug=slug_input)
 
-# run query as per user input
+    query = slug_response.id
 
-i = 1
-while i <= times :
-    response = sdk.run_query(
-    query_id=query,
-    result_format="json_detail",
-    cache=False,
-    cache_only=False)
-    # Convert string to Python dict 
-    query_dic = json.loads(response) 
-    runtime_duration = round(float(query_dic['runtime']), 2)
-    ran_at_utc = pytz.utc.localize(datetime.fromisoformat(query_dic['ran_at'][:-1]))
-    ran_at = ran_at_utc.astimezone(pytz.timezone('America/Vancouver')).isoformat()
-    query_list.append(runtime_duration)
     if args.file:
-        with open(filename, 'a', encoding='UTF8') as f:
+        with open(filename, 'w', encoding='UTF8') as f:
             writer = csv.writer(f)
-            # write the data
-            writer.writerow([slug_input, ran_at, runtime_duration])
-    
-    # adding 5 min time delay between subsequent queries to keep looker open for other tasks
-    if i != times:
-        time.sleep(sleep_timer)
-
-    i += 1
-
-if args.file:
-    print("Filename =", filename)
-    f.close()
-    
-#Print for Console 
-
-print("RunTimes =", query_list)
-minimum = min(query_list)
-print("Min =", minimum)
-maximum = max(query_list)
-print("Max =", maximum)
-sum = sum(query_list)
-length = len(query_list)
-average = round(sum/length, 2)
-print("Avg =", average)
+            # write the header
+            writer.writerow(header)
+
+    # run query as per user input
+
+    i = 1
+    while i <= times :
+        response = sdk.run_query(
+        query_id=query,
+        result_format="json_detail",
+        cache=False,
+        cache_only=False)
```

Comment:
You can have the `ALTER USER` statements on either side of this single call.

Doing so would let the Looker user continue to have have cache access, and only disable cache access in the one moment where the Looker API is called and and a response received.

So if you had a 100 second delay, anyone querying through the Looker user could still take advantage of being able to use the Redshift cache through that user's access to it.

Reply:
Moved please re-review

Reply:
please put this into a try block and except on a `sdk_error.SDKError`.

Inside that exception you need to execute the `ALTER USER looker SET enable_result_cache_for_session TO on;` and then `exit(1)`.

e.g.:
```
try:
        response = sdk.run_query(
        query_id=query,
        result_format="json_detail",
        cache=False,
        cache_only=False)
except error.SDKError as err:
       print(f'Exiting due to Looker SDK exception: {err}'
       with conn:
            with conn.cursor() as curs:
                try:
                       curs.execute("ALTER USER looker SET enable_result_cache_for_session TO on;")
                except Exception as err:
                       print(f'URGENT! Exiting due to psycopg2 execution error: {err}')
                       print('Requires manual run of: ALTER USER looker SET enable_result_cache_for_session TO on;')
                       exit(1)
       exit(1)
```
It's important that the user running the script know that they _have_ to run the SQL `ALTER USER` statement if an exception occurred when trying to run `SET enable_result_cache_for_session TO on` from this script.

per: https://github.com/looker-open-source/sdk-codegen/blob/main/python/looker_sdk/error.py